common/src/com/thoughtworks/go/domain/materials/AgentSubprocessExecutionContext.java

```diff
@@ -40,4 +40,9 @@ public String getProcessNamespace(String fingerprint) {
     public Map<String, String> getDefaultEnvironmentVariables() {
         return new HashMap<>();
     }
+
+    @Override
+    public Boolean getGitShallowCloneFlag() {
+        throw new UnsupportedOperationException();
```

Comment:
Because agents SHOULD NOT be calling this method, and if they are, there's something really-really wrong. We perhaps need a better exception class, or a better error message `throw new RuntimeException("Really useful message")`?